add istioctl install page for ambient and improve install docs experience

craigbox opened the pull request:
Add docs on how to install ambient with istioctl.
When @bleggett's changes to Helm docs are in, I'll modify ths further to bring the two in line.

- [x] Ambient
- [x] Docs

istio-testing commented:
@craigbox: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | 5b7e02514326b63d654aa2bec0db510412ae2390 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/lint_istio.io/1817852275062738944) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

dhawton commented on content/en/docs/ambient/install/istioctl/index.md:
```diff
@@ -0,0 +1,80 @@
+[quick start instructions](/docs/ambient/getting-started) instead.
+{{< /tip >}}
+
+This installation guide uses the [istioctl](/docs/reference/commands/istioctl/) command line
+tool to provide rich customization of Istio installation options
+It has user input validation to help prevent installation errors and customization options to
+override any aspect of the configuration.
```
```suggestion
This installation guide uses the [istioctl](/docs/reference/commands/istioctl/) command-line
tool to provide rich customization of Istio installation options. It has user input validation
to help prevent installation errors and customization options to override any aspect of the
configuration.
```

dhawton commented on content/en/docs/ambient/install/istioctl/index.md:
```diff
@@ -0,0 +1,80 @@
+
+## Install Istio using the ambient profile
+
+`istioctl` supports a number of [configuration profiles](/docs/setup/additional-setup/config-profiles/) that include different default options, and can be customized for your production needs. Support for ambient mode is included in the `ambient` profile. Install Istio with the following command:
```
```suggestion
`istioctl` supports a number of [configuration profiles](/docs/setup/additional-setup/config-profiles/) that include different default options,
and can be customized for your production needs. Support for ambient mode is included in the `ambient` profile. Install Istio with the
following command:
```

dhawton commented on content/en/docs/ambient/install/istioctl/index.md:
```diff
@@ -0,0 +1,80 @@
+
+## Configure and modify profiles
+
+Istio's installation API is documented in the [`IstioOperator` API reference](/docs/reference/config/istio.operator.v1alpha1/). You can use the `--set` option to `istioctl install` to modify individual installation parameters, or specify your own configuration file with `-f`.
```
```suggestion
Istio's installation API is documented in the [`IstioOperator` API reference](/docs/reference/config/istio.operator.v1alpha1/). You
can use the `--set` option to `istioctl install` to modify individual installation parameters, or specify your own configuration file with `-f`.
```

dhawton commented on content/en/docs/ambient/install/istioctl/index.md:
```diff
@@ -0,0 +1,80 @@
+weight: 10
+keywords: [istioctl,ambient]
+owner: istio/wg-environments-maintainers
+test: no
```
```suggestion
test: yes
```

craigbox replied:
It doesn't really do anything, so I don't think testing adds a lot of value?  Do we need the snips for something else?

dhawton replied:
I feel like anything we have a command that has commands on it, we should test it so that we can catch breakages from istio/istio, unless the feature is experimental. Save ourselves from having parts of docs that will be wrong if changes occur.

craigbox replied:
a noble goal

dhawton commented on content/en/docs/ambient/install/platform-prerequisites/index.md:
```diff
@@ -1,73 +1,77 @@
+
+If you are installing Istio on [MicroK8s](https://microk8s.io/), you must append `--set values.cni.cniConfDir=/var/snap/microk8s/current/args/cni-network --set values.cni.cniBinDir=/var/snap/microk8s/current/opt/cni/bin` to the `helm install` command, as MicroK8s [uses non-standard locations for CNI configuration and binaries](https://microk8s.io/docs/change-cidr).
+
+### Minikube
```
```suggestion
### minikube
```

dhawton commented on content/en/docs/ambient/install/platform-prerequisites/index.md:
```diff
@@ -1,73 +1,77 @@
+
+### Minikube
+
+If you are using [Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/) with the [Docker driver](https://minikube.sigs.k8s.io/docs/drivers/docker/),
```
```suggestion
If you are using [minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/) with the [Docker driver](https://minikube.sigs.k8s.io/docs/drivers/docker/),
```

dhawton commented on content/en/docs/ambient/install/platform-prerequisites/index.md:
```diff
@@ -1,73 +1,77 @@
     $ istioctl install --set profile=ambient --skip-confirmation  --set values.cni.cniConfDir=/var/lib/rancher/k3s/agent/etc/cni/net.d --set values.cni.cniBinDir=/bin
     {{< /text >}}
 
 ### K3S
```
```suggestion
### K3s
```

dhawton commented on content/en/docs/ambient/install/platform-prerequisites/index.md:
```diff
@@ -1,73 +1,77 @@
 ### K3S
 
-1. If you are using [K3S](https://k3s.io/) and one of its bundled CNIs, you must append `--set values.cni.cniConfDir=/var/lib/rancher/k3s/agent/etc/cni/net.d --set values.cni.cniBinDir=/var/lib/rancher/k3s/data/current/bin/` to your `istioctl install` or `helm install` command to install Istio ambient, as K3S uses nonstandard locations for CNI configuration and binaries. These nonstandard locatio ...
+When using [K3S](https://k3s.io/) and one of its bundled CNIs, you must append `--set values.cni.cniConfDir=/var/lib/rancher/k3s/agent/etc/cni/net.d --set values.cni.cniBinDir=/var/lib/rancher/k3s/data/current/bin/` to your `istioctl install` or `helm install` command to install Istio ambient, as K3S uses nonstandard locations for CNI configuration and binaries. These nonstandard locations may be ...
```
```suggestion
When using [K3s](https://k3s.io/) and one of its bundled CNIs, you must append `--set values.cni.cniConfDir=/var/lib/rancher/k3s/agent/etc/cni/net.d --set values.cni.cniBinDir=/var/lib/rancher/k3s/data/current/bin/` to your `istioctl install` or `helm install` command to install Istio ambient, as K3S uses nonstandard locations for CNI configuration and binaries. These nonstandard locations may be overridden as well [according to K3s documentation](https://docs.k3s.io/cli/server#k3s-server-cli-help). If you are using K3s with a custom, non-bundled CNI, you must use the correct paths for those CNIs, e.g. `/etc/cni/net.d` - [see K3s docs for details](https://docs.k3s.io/networking/basic-network-options#custom-cni).
```

dhawton commented on content/en/docs/ambient/install/platform-prerequisites/index.md:
```diff
@@ -1,73 +1,77 @@
+      - system-node-critical
+{{< /text >}}
 
 ### K3D
```
```suggestion
### k3d
```

istio-testing commented:
@craigbox: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | 27d8e611d98f5cf313e315e8b4558b460759756c | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/lint_istio.io/1818211589543694336) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

bleggett commented on content/en/docs/ambient/install/istioctl/index.md:
```diff
@@ -0,0 +1,83 @@
+
+Before you begin, check the following prerequisites:
+
+1. [Download the Istio release](/docs/setup/additional-setup/download-istio-release/).
```
1. I don't think we should tell people to "download the Istio release" - we should probably just tell them to 

1. Install `istioctl` (which is currently buried -> https://istio.io/latest/docs/ops/diagnostic-tools/istioctl/#install-hahahugoshortcode686s2hbhb - we should maybe promote that from `diagnostic-tool-only setup` to `proper istioctl setup`.

2. Document the install from that. `istioctl` binary by itself should probably be enough to bootstrap, if the goal here is "istioctl is the simple, single entry point".

craigbox replied:
This is a direct lift from the sidecar `install with istioctl page.

The "download release" page was separated out from the sidecar getting started docs, in order to document the options without putting them in front of people on day 1.  It's a weird wording but remember "Istio release" is "istioctl plus samples".  We use it throughout the docs at present.

We could rename it "download client" and condense these two pages?

(and maybe even tell people you can use `brew` or apt/rpm)

bleggett replied:
Can we just tell people to install `istioctl` here using the same instructions as the "install Istioctl page"?

Maybe even link to that page for further context/

https://preliminary.istio.io/latest/docs/ops/diagnostic-tools/istioctl/#install-hahahugoshortcode708s2hbhb

That is, the script that doesn't require downloading the samples.

We should be asking people to download the samples on the page where they actually need the samples, which is not this page.

And we should be using the same install script consistently everywhere.

craigbox replied:
https://github.com/istio/istio.io/issues/15585

istio-testing commented:
@craigbox: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
gencheck_istio.io | c460a34041e8905e14f318ac47f8992c5376eeb7 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/gencheck_istio.io/1825384774689099776) | true | `/test gencheck`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@craigbox: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | cbb1f1994a7aca337c4e89a23eb32cd6d55d77e6 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/lint_istio.io/1825385535099637760) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@craigbox: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | cbb1f1994a7aca337c4e89a23eb32cd6d55d77e6 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/lint_istio.io/1825385535099637760) | true | `/test lint`
doc.test.profile-none_istio.io | cbb1f1994a7aca337c4e89a23eb32cd6d55d77e6 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-none_istio.io/1825385532557889536) | true | `/test doc.test.profile-none`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@craigbox: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | 9b6931c7a81a56b53848d3d54fb61e09fac09256 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/lint_istio.io/1825433272700637184) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@craigbox: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | 9b6931c7a81a56b53848d3d54fb61e09fac09256 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/lint_istio.io/1825433272700637184) | true | `/test lint`
gencheck_istio.io | a7f86708e2abace1c084c217f37512f219a52d96 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/gencheck_istio.io/1825479654836801536) | true | `/test gencheck`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@craigbox: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
doc.test.profile-demo_istio.io | 16dc1a684ddeb9e82543fb3ab24363de7358aded | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-demo_istio.io/1825487219540889600) | true | `/test doc.test.profile-demo`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@craigbox: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
doc.test.profile-demo_istio.io | 16dc1a684ddeb9e82543fb3ab24363de7358aded | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-demo_istio.io/1825487219540889600) | true | `/test doc.test.profile-demo`
doc.test.profile-default_istio.io | 16dc1a684ddeb9e82543fb3ab24363de7358aded | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-default_istio.io/1825487219477975040) | true | `/test doc.test.profile-default`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

bleggett commented on content/en/docs/ambient/install/platform-prerequisites/index.md:
```diff
@@ -1,73 +1,77 @@
-### K3S
+### K3s
+
+When using [K3s](https://k3s.io/) and one of its bundled CNIs, you must append `--set values.cni.cniConfDir=/var/lib/rancher/k3s/agent/etc/cni/net.d --set values.cni.cniBinDir=/var/lib/rancher/k3s/data/current/bin/` to your `istioctl install` or `helm install` command to install Istio ambient, as K3S uses nonstandard locations for CNI configuration and binaries. These nonstandard locations may be ...
```
Note that because `istioctl` is weird, `--sets` are not cross-compatible between Helm and `istioctl` which will catch copypasters.

- For helm it's `--set cni.cniConfDir=/var/lib/rancher/k3s/agent/etc/cni/net.d`
- For istioctl it's `--set values.cni.cniConfDir=/var/lib/rancher/k3s/agent/etc/cni/net.d`

bleggett commented on content/en/docs/ambient/install/platform-prerequisites/index.md:
```diff
@@ -1,73 +1,77 @@
+
+### MicroK8s
+
+If you are installing Istio on [MicroK8s](https://microk8s.io/), you must append `--set values.cni.cniConfDir=/var/snap/microk8s/current/args/cni-network --set values.cni.cniBinDir=/var/snap/microk8s/current/opt/cni/bin` to the `helm install` command, as MicroK8s [uses non-standard locations for CNI configuration and binaries](https://microk8s.io/docs/change-cidr).
```
All of these also apply to `istioctl`, see also above comment about `--set`

istio-testing commented:
@craigbox: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
doc.test.profile-demo_istio.io | 16dc1a684ddeb9e82543fb3ab24363de7358aded | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-demo_istio.io/1825487219540889600) | true | `/test doc.test.profile-demo`
doc.test.profile-default_istio.io | 16dc1a684ddeb9e82543fb3ab24363de7358aded | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-default_istio.io/1825487219477975040) | true | `/test doc.test.profile-default`
gencheck_istio.io | dee02ef33ca15e979ed34a209c5aa228c3729a8d | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/gencheck_istio.io/1825680588875501568) | true | `/test gencheck`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@craigbox: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
doc.test.profile-demo_istio.io | 16dc1a684ddeb9e82543fb3ab24363de7358aded | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-demo_istio.io/1825487219540889600) | true | `/test doc.test.profile-demo`
gencheck_istio.io | dc78c144827afb7789faba47aeab1e31a4ab1341 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/gencheck_istio.io/1825681957720166400) | true | `/test gencheck`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@craigbox: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
doc.test.profile-demo_istio.io | 16dc1a684ddeb9e82543fb3ab24363de7358aded | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-demo_istio.io/1825487219540889600) | true | `/test doc.test.profile-demo`
gencheck_istio.io | dc78c144827afb7789faba47aeab1e31a4ab1341 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/gencheck_istio.io/1825681957720166400) | true | `/test gencheck`
lint_istio.io | dc78c144827afb7789faba47aeab1e31a4ab1341 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/lint_istio.io/1825681958546444288) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@craigbox: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
doc.test.profile-none_istio.io | b90236533e7567bab5902a3255e10a88e1694322 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-none_istio.io/1825685604252258304) | true | `/test doc.test.profile-none`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@craigbox: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
doc.test.profile-none_istio.io | b90236533e7567bab5902a3255e10a88e1694322 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-none_istio.io/1825685604252258304) | true | `/test doc.test.profile-none`
doc.test.profile-default_istio.io | ac30af39c129af3d4613144937e409fabf92cb7d | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-default_istio.io/1825705272044163072) | true | `/test doc.test.profile-default`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@craigbox: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
doc.test.profile-default_istio.io | ac30af39c129af3d4613144937e409fabf92cb7d | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-default_istio.io/1825705272044163072) | true | `/test doc.test.profile-default`
doc.test.profile-none_istio.io | ac30af39c129af3d4613144937e409fabf92cb7d | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-none_istio.io/1825705272207740928) | true | `/test doc.test.profile-none`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@craigbox: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
doc.test.profile-none_istio.io | ac30af39c129af3d4613144937e409fabf92cb7d | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-none_istio.io/1825705272207740928) | true | `/test doc.test.profile-none`
doc.test.profile-default_istio.io | 36bcf8ae980b8f557e4776d3fa15980680a68070 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-default_istio.io/1825739455080697856) | true | `/test doc.test.profile-default`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

craigbox commented:
/test doc.test.profile-default

(#15576)

istio-testing commented:
@craigbox: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
doc.test.profile-default_istio.io | 36bcf8ae980b8f557e4776d3fa15980680a68070 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-default_istio.io/1825832801618890752) | true | `/test doc.test.profile-default`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

bleggett commented on content/en/docs/ambient/install/platform-prerequisites/index.md:
```diff
@@ -1,73 +1,178 @@
+
+{{< /tabset >}}
+
+### Red Hat OpenShift
```
Good, ty!

bleggett commented on content/en/docs/ambient/install/istioctl/index.md:
```diff
@@ -0,0 +1,83 @@
+
+## Install Istio using the ambient profile
+
+`istioctl` supports a number of [configuration profiles](/docs/setup/additional-setup/config-profiles/) that include different default options,
+and can be customized for your production needs. Support for ambient mode is included in the `ambient` profile. Install Istio with the
+following command:
```
```suggestion
Support for ambient mode is included in the `ambient` profile. Install Istio with the
following command:
```

The `profiles` support between `istioctl` and `helm` is 1-to-1 - they are the same profiles, in fact.

So either we need to blurb this on the helm page too ("hey, we have other profiles, go look") - or do what that page does, and don't blurb it.

Either way, should be consistent.

craigbox replied:
The sidecar installation docs haven't been meaningfully updated from the time when they were the only installation method.

I agree that a complete overhaul of this is a good idea but I'm not going to boil the ocean in this PR.  I haven't created an issue for this one because I believe that https://github.com/istio/istio/issues/52702 should be general enough for sidecars too.

craigbox replied:
On the topic of ambient specifically, with istioctl you need `profile=ambient` in order to tell it to install ztunnel and istio-cni.  It's reasonable to point that out here - if you only did `istioctl install`, you wouldn't get working ambient mode. 

Given that, in the Helm docs, we explicitly say you need to install Helm charts A B C and D, individually including ztunnel and CNI. What use are profiles there?  Why are these two lines different?

```
$ helm install istiod istio/istiod --namespace istio-system --set profile=ambient --wait
$ helm install ztunnel istio/ztunnel -n istio-system --wait 
```

craigbox replied:
OK, it's explained in https://github.com/istio/istio/blob/master/manifests/helm-profiles/ambient.yaml that they are just sets of values, not groups of components. 

an IstioOperator profile is a list of components and a link to a Helm profile.

this isn't half confusing.

bleggett replied:
Profiles for both are almost entirely (and primarily) sets of values, and in fact both the Istio and helm ones are derived from the same file IIRC.

(they could be both sets of values and of components for both helm and istioctl pretty easily if we did chart composition in helm, but we do not).

craigbox replied:
[The ambient Helm profile](https://github.com/istio/istio/blob/master/manifests/helm-profiles/ambient.yaml) contains environment variables and flags.

[The ambient istioctl profile](https://github.com/istio/istio/blob/master/manifests/profiles/ambient.yaml) contains a list of components to enable, and an instruction to Helm to use the ambient Helm profile.

So yes, you're mostly right, but in the specific case of ambient, the istioctl profile is necessary to say "install ztunnel", where that is spelled out in the Helm docs. 

This has been a useful exercise to understand the distinction, and if we could get chart composition - I assume you get that through [dependencies](https://helm.sh/docs/topics/charts)? - then it would be possible to get the same outcome yes, though not through something called a profile. :)

howardjohn replied:
I doubt the helm profiles would work via dependencies unfortunately... helm doesn't have native 'profile' concept so we hack it in a way that would apply the config _after_ it has chosen which dependencies to enable I suspect (untested) (https://blog.howardjohn.info/posts/advanced-helm/)

bleggett replied:
Dependencies in helm can be enabled or disabled based on `values.yaml` properties, so I would _expect_ if you disabled a component but a profile still supplied values for it, that would just no-op in the templating and give you ~roughly the same behavior as `istioctl install <specific component> -f large-operator-yaml-with-many-settings-that-do-not-apply-to-that-component.yaml`

but yeah, need to test.

bleggett commented on content/en/docs/ambient/install/istioctl/index.md:
```diff
@@ -0,0 +1,83 @@
+to help prevent installation errors and customization options to override any aspect of the
+configuration.
+
+Using these instructions, you can select any one of Istio's built-in
+[configuration profiles](/docs/setup/additional-setup/config-profiles/)
+and then further customize the configuration for your specific needs.
```
Ditto - either advertise the same profile stuff on the Helm page (this is not unique to `istioctl`, the same profiles are used for both options) or drop it in both places.

craigbox replied:
This will need to be cleaned up in the context of IstioOperator no longer being a custom resource.

bleggett commented on content/en/docs/ambient/install/istioctl/index.md:
```diff
@@ -0,0 +1,83 @@
+
+## Configure and modify profiles
+
+Istio's installation API is documented in the [`IstioOperator` API reference](/docs/reference/config/istio.operator.v1alpha1/). You
+can use the `--set` option to `istioctl install` to modify individual installation parameters, or specify your own configuration file with `-f`.
```
If we tell people how to do this in the `istioctl` install doc, we should tell them how to do (roughly exactly the same thing) in the Helm doc, or remove this.

i.e with Helm it's `-f values-overriddes.yaml` or `--set` as well, if we mention it here we should mention it there, or in neither spot.

craigbox replied:
I've copied some text from https://istio.io/latest/docs/setup/install/helm/ to that effect.

bleggett commented on content/en/docs/ambient/upgrade/helm/index.md:
```diff
@@ -80,34 +83,36 @@ If you have customized your istiod installation, you can reuse the `values.yaml`
 
 {{< warning >}}
-Upgrading the Istio CNI agent to a compatible version in-place will not disrupt networking for running pods already successfully added to an ambient mesh, but no ambient-captured pods will be successfully scheduled (or rescheduled) on the node until the upgrade is complete and the upgraded Istio CNI agent on the node passes readiness checks. If this is a significant disruption concern, or stricte ...
+Upgrading ztunnel in-place will briefly disrupt all ambient mesh traffic on the node.
```
```suggestion
Upgrading ztunnel in-place will briefly disrupt all ambient mesh traffic on the node, regardless of the use of revisions. In practice the disruption period is a very small window, primarily affecting long-running connections.
```

istio-testing commented:
@craigbox: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
doc.test.profile-default_istio.io | 36bcf8ae980b8f557e4776d3fa15980680a68070 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-default_istio.io/1825832801618890752) | true | `/test doc.test.profile-default`
lint_istio.io | 11d75fadca83c7ce3ec05894169fc07447264c4e | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/lint_istio.io/1826047757035507712) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

craigbox commented on content/en/docs/ambient/install/platform-prerequisites/index.md:
```diff
@@ -1,73 +1,178 @@
+## CNI plugins
 
-## CNI
+The following configurations apply to all platforms, when certain {{< gloss "CNI" >}}CNI plugins{{< /gloss >}} are used:
```
It would be more correct to say "[CNI runtime](https://github.com/containernetworking/cni/blob/main/SPEC.md#overview)"?  But less understood, I would think. [Kubernetes uses the term 'CNI plugin'](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/), as do many of the people that make them:

> Kubernetes uses the Container Network Interface (CNI) to interact with networking providers like Calico. The Calico binary that presents this API to Kubernetes is called the CNI plugin and must be installed on every node in the Kubernetes cluster.

and thus I think this is a fight that is lost outside of the universe of people who actually write CNI implementations.

We were trying to have our thing be called the 'CNI node agent' to differentiate it.  Is that not sensible?

(p.s. I see you)

<img width="1454" alt="Screenshot 2024-08-21 at 12 16 44 PM" src="https://github.com/user-attachments/assets/98400170-ed08-4767-82d9-b237011a1545">

bleggett replied:
`CNI node agent` is the most sensible, I mostly just don't want to say `CNI plugin` and have it not be clear if we're talking about our thing (which is not a CNI, and consists of more than a plugin) or the thing you already have (which is, and consists of more than a plugin).

craigbox commented on content/en/docs/ambient/install/_index.md:
```diff
@@ -1,10 +1,18 @@
+{{< /tip >}}
+
+There are two different options for installing Istio. Which one you should choose will depend on your production requirements, and how you install other software. If you need help choosing, refer to our
+[which Istio installation method should I use?](/about/faq/#install-method-selection) FAQ page.
```
https://github.com/istio/istio.io/issues/15584

craigbox commented on content/en/docs/ambient/install/helm/index.md:
```diff
@@ -1,17 +1,23 @@
+[quick start instructions](/docs/ambient/getting-started) instead.
+{{< /tip >}}
+
+We encourage the use of Helm to install Istio for use in ambient mode. To allow controlled upgrades, the control plane and data plane components are packaged and installed separately. (Because the ambient data plane is split across [two components](/docs/ambient/architecture/data-plane), the ztunnel and waypoints, upgrades involve separate steps for these components.)
```
https://github.com/istio/istio.io/issues/15584 again, though that FAQ doesn't know ambient exists, and I think it's actually possible the recommendation is istioctl for sidecars and Helm for ambient.  We'll let that process play out.

craigbox replied:
https://github.com/istio/istio.io/issues/15584 again, though that FAQ doesn't know ambient exists, and it may even be the case the recommendation is istioctl for sidecars and Helm for ambient.  We'll let that process play out.

craigbox commented on content/en/docs/ambient/install/istioctl/index.md:
```diff
@@ -0,0 +1,83 @@
+[quick start instructions](/docs/ambient/getting-started) instead.
+{{< /tip >}}
+
+This installation guide uses the [istioctl](/docs/reference/commands/istioctl/) command-line
+tool to provide rich customization of Istio installation options. It has user input validation
+to help prevent installation errors and customization options to override any aspect of the
+configuration.
+
```
rewritten a bit and accepted

istio-testing commented:
@craigbox: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
gencheck_istio.io | 8dce7528df6fc50b323a72cdd5b2dd3a11f68c6b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/gencheck_istio.io/1826071619236270080) | true | `/test gencheck`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

craigbox commented:
I've addressed all of Ben's comments with a bunch of "will fix later" issues, and what would you know, I think I've fixed #15576.

@dhawton for approval, or telling me to split the fix into its own PR first.

kfaseela commented:
LGTM

howardjohn commented on content/en/docs/ambient/install/platform-prerequisites/index.md:
```diff
@@ -1,73 +1,178 @@
-1. If you are using [Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/) with the [Docker driver](https://minikube.sigs.k8s.io/docs/drivers/docker/),
-you must append `--set cni.cniNetnsDir="/var/run/docker/netns"` to the `helm install` command so that the `istio-cni` node agent can correctly manage
-and capture pods on the node.
+On GKE, Istio components with the [system-node-critical](https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/) `priorityClassName` can only be installed in namespaces that have a [ResourceQuota](https://kubernetes.io/docs/concepts/policy/resource-quotas/) defined. By default in GKE, only `kube-system` has a defined ResourceQuota for the `node-critical` cl ...
```
off topic but we should automate this or warn in istioctl

howardjohn commented on content/en/docs/ambient/install/platform-prerequisites/index.md:
```diff
@@ -1,73 +1,178 @@
+{{< /text >}}
 
-### K3D
+### k3d
```
also off topic, we should automagically detect this, k3s, and microk8s

howardjohn reviewed (commented):
LGTM

istio-testing commented:
@craigbox: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
doc.test.profile-demo_istio.io | 66efa886d2e81da68336214bced9533d7f21b5b1 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15483/doc.test.profile-demo_istio.io/1826339081089454080) | unknown | `/test doc.test.profile-demo`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

craigbox commented:
/test doc.test.profile-demo

craigbox commented:
/cherry-pick release-1.23
